packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -639,9 +639,9 @@ export class AppSyncModelVisitor<
   }
 
   protected generateIntermediateModel(firstModel: CodeGenModel, secondModel: CodeGenModel, relationName: string) {
-    const firstModelKeyFieldName = `${camelCase(firstModel.name)}ID`;
+    const firstModelKeyFieldName = this.generateIntermediateModelPrimaryKeyFieldName(firstModel)
     const firstModelSortKeyFields: CodeGenField[] = this.getSortKeyFields(firstModel);
-    const secondModelKeyFieldName = `${camelCase(secondModel.name)}ID`;
+    const secondModelKeyFieldName = this.generateIntermediateModelPrimaryKeyFieldName(secondModel);
```

Comment:
Are these field name changes behind a feature flag? I can see this breaking applications if we suddenly change the column that data is being stored in dynamo, as well as if any customers are making references to these fields in their code

Reply:
Yes it is behind feature flag `respectPrimaryKeyAttributesOnConnectionField` which is the same on transformer side